### PR TITLE
feat: account management

### PR DESCRIPTION
**Pull request by bradleySuira**
## What?
- Account Creation and management including the option to enable/disable blockchains inside the wallet

#Notion/ Trello
- [https://www.notion.so/Implement-Multi-Account-Management-in-UI-2febb3da21214cb3bae5eb086805ed8a](https://www.notion.so/Implement-Multi-Account-Management-in-UI-2febb3da21214cb3bae5eb086805ed8a)
- [https://www.notion.so/Implement-Multi-Account-Creation-in-UI-b03da731da734d4ea1d565bd12005e68](https://www.notion.so/Implement-Multi-Account-Creation-in-UI-b03da731da734d4ea1d565bd12005e68)

## Testing?
- [ ] Run tests locally

## Screenshots (optional)
0

### CODE REVIEW

File: src/views/Swap/Swap.vue [hunk 1]
```diff
@@ -1020,6 +1023,18 @@ export default {
...
       this.bridgeModalOpen = false
+    },
+    trackNoLiquidity () {
+      if (this.showNoLiquidityMessage) {
+        this.trackAnalytics({
+          event: 'No Liquidity',
+          properties: {
+            category: 'Swap screen',
+            action: 'No Liquidity for pairs',
+            label: `No Liquidity for ${this.asset} to ${this.toAsset}`
```
Comment (beemi) [inline src/views/Swap/Swap.vue:1034]: @bradleySuira  can u change label to 

`fron:  ${this.asset}`,
`to: ${this.toAsset}`

**Review by beemi — CHANGES_REQUESTED** [review]

File: src/views/Swap/Swap.vue (hunk 1 above)
Comment (bradleySuira) [reply src/views/Swap/Swap.vue]: done mate

File: src/store/mutations.js [hunk 2]
```diff
@@ -274,5 +274,41 @@ export default {
...
+  TOOGLE_BLOCKCHAIN (state, { network, walletId, chainId, enable }) {
+    if (!state.enabledChains) {
+      Vue.set(state, 'enabledChains', {})
+    }
+    if (!state.enabledChains[walletId]) {
+      Vue.set(state.enabledChains, walletId, {})
+    }
+    if (!state.enabledChains[walletId]?.[network]) {
+      Vue.set(state.enabledChains[walletId], network, [])
+    }
```
Comment (monokh) [inline src/store/mutations.js:287]: Can the 3 above be replaced with something like `ensureNetworkWalletTree` ?

File: src/store/mutations.js [hunk 3]
```diff
@@ -274,5 +274,41 @@ export default {
...
+
+    const chains = state.enabledChains.[walletId].[network]
+    if (enable) {
+      Vue.set(state.enabledChains.[walletId], network, [...new Set([...chains, chainId])])
+    } else {
+      Vue.set(state.enabledChains.[walletId], network, [...new Set([...chains.filter(c => c !== chainId)])])
+    }
+  },
+  TOOGLE_ACCOUNT (state, { network, walletId, accountId, enable }) {
+    const accounts = state.accounts[walletId][network]
```
Comment (monokh) [inline src/store/mutations.js:297]: Does it need `ensureNetworkWalletTree` here too?

File: src/store/mutations.js [hunk 4]
```diff
@@ -274,5 +274,41 @@ export default {
...
+    if (accounts) {
```
Comment (monokh) [inline src/store/mutations.js:298]: should be more strict, like `Array.isArray(accounts) && accounts.length`

File: src/store/mutations.js [hunk 5]
```diff
@@ -274,5 +274,41 @@ export default {
...
+      const index = accounts.findIndex(
+        (a) => a.id === accountId
+      )
+
+      if (index >= 0) {
+        const _account = accounts[index]
+        const updatedAccount = {
+          ..._account,
+          enabled: !!enable
```
Comment (monokh) [inline src/store/mutations.js:307]: Why `!!enable` ? It should always be passed a boolean. It sounds worrying that you need to do this, is there a bug somewhere?

File: src/views/Accounts/Create.vue [hunk 6]
```diff
@@ -1,140 +1,325 @@
...
+        enabled: true,
+        color: this.accountColor
+      }
+      const createdAccount = await this.createAccount({
+        network: this.activeNetwork,
+        walletId: this.activeWalletId,
+        account
+      })
+
+      await this.getUnusedAddresses({
```
Comment (monokh) [inline src/views/Accounts/Create.vue:245]: Why not put the updating addresses and balances inside the `createAccount` action?

**Review by monokh — CHANGES_REQUESTED** [review]
image
Chain order in decending alphabetical doesn't seem right

Turned off bitcoin chain, went to swap, it looked like this:
image
Maybe its defaulting to chain that is not enabled

When i add account, the liq logo looks weird
image

anyway why you need to enter password to add account? :S
I didn't enter password, it was still added to my accounts after i open the wallet again.

The steps were:

Add account for RSK
Change chain to ETH
Change chain back to RSK
Click create

File: src/store/actions/accounts/toogleAccount.js [hunk 7]
```diff
@@ -0,0 +1,3 @@
+export const toogleAccount = async ({ commit }, { network, walletId, accountId, enable }) => {
```
[the remark this answers is not in the document]
Comment (bradleySuira) [reply src/store/actions/accounts/toogleAccount.js]: fixed

File: src/store/mutations.js (hunk 2 above)
Comment (bradleySuira) [reply src/store/mutations.js:287]: fixed

File: src/store/mutations.js (hunk 3 above)
Comment (bradleySuira) [reply src/store/mutations.js:308]: added

File: src/store/mutations.js (hunk 5 above)
Comment (bradleySuira) [reply src/store/mutations.js:318]: fixed, should be a boolean value always

File: src/views/Accounts/Create.vue (hunk 6 above)
Comment (bradleySuira) [reply src/views/Accounts/Create.vue]: fixed, now calling required actions inside `createAcction` action

File: src/store/mutations.js (hunk 4 above)
Comment (bradleySuira) [reply src/store/mutations.js:309]: now we are using a different approach also ensuring that we have the account tree created

**Pull request merged**